newsfragments/46375.significant.rst

```diff
@@ -0,0 +1,14 @@
+``SecretsMasker`` has now been moved into the task SDK to be consumed by DAG authors and users
+
+Any occurrences of the ``secrets_masker`` module will have to be updated from ``airflow.utils.log.secrets_masker`` to the new path: ``airflow.sdk.execution_time.secrets_masker``
+
+* Types of change
+
+  * [ ] Dag changes
+  * [ ] Config changes
+  * [ ] API changes
+  * [ ] CLI changes
+  * [x] Behaviour changes
+  * [ ] Plugin changes
+  * [ ] Dependency changes
+  * [ ] Code interface changes
```

Comment:
We'll need migration rules for it. Looks like this is path change. We'll need an AIR302 ruff rule. You can follow the example here

https://github.com/apache/airflow/blob/699171a6d455cfd66ecc81fd5638129d06317826/newsfragments/41395.significant.rst?plain=1#L23-L33

Reply:
Thanks Wei, yeah it is a path change

Reply:
Just pushed a fix for it